src/components/UserPofile/UserProfile.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { FormEvent, useEffect } from "react";
+import { useRouter } from "next/router";
+import { useAuthStore } from "@/store/auth";
+import { Button } from "../Controls/Button";
+import { InputMasked } from "../Input/InputMasked";
+import { Preloader } from "../Preloader/Preloader";
+import style from "./UserCabinet.module.css";
+import classNames from "classnames";
+import { useUpdateProfile } from "../../hooks/useProfileUpdate";
+import { useFileUpload } from "../../hooks/useFileUpload";
+import { usePhoneNumber } from "../../hooks/usePhoneNumber";
+
+export const UpdateProfile: React.FC = () => {
+  const { user } = useAuthStore();
+  const router = useRouter();
+
+  const { loading, successMessage, errorMessage, handleUpdateProfile } = useUpdateProfile(user);
+  const { phoneNumber, handlePhoneNumberChange } = usePhoneNumber(user?.phoneNumber || "");
+  const { fileInputRef, fileName, imageError, handleFileInputChange, handleUpload, getShortFileName } = useFileUpload();
+
+  const handleSubmit = async (event: FormEvent) => {
+    event.preventDefault();
+    await handleUpdateProfile(phoneNumber);
+  };
+
+  useEffect(() => {
```

Comment:
this logic should be in user profile page not in form, it's not responsibility of this form to redirect